lib/commons/color/get-background-color.js

```diff
@@ -176,236 +284,87 @@ function includeMissingElements(elmStack, elm) {
-		getBgColor(document.documentElement).alpha === 0
-	) {
-		// Remove body and html from it's current place
-		bgNodes.splice(bodyIndex, 1);
-		bgNodes.splice(elmStack.indexOf(document.documentElement), 1);
-
-		// Put the body background as the lowest element
-		bgNodes.push(document.body);
+function elmPartiallyObscured(elm, bgElm, bgColor) {
+	var obscured =
+		elm !== bgElm && !dom.visuallyContains(elm, bgElm) && bgColor.alpha !== 0;
+	if (obscured) {
+		axe.commons.color.incompleteData.set('bgColor', 'elmPartiallyObscured');
 	}
-	return bgNodes;
+	return obscured;
 }
-/**
- * Get coordinates for an element's client rects or bounding client rect
- * @method getCoords
- * @memberof axe.commons.color
- * @instance
- * @param {DOMRect} rect
- * @return {Object}
- */
-color.getCoords = function(rect) {
-	let x, y;
-	if (rect.left > window.innerWidth) {
-		return;
-	}
-	if (rect.top > window.innerHeight) {
-		return;
-	}
-	x = Math.min(Math.ceil(rect.left + rect.width / 2), window.innerWidth - 1);
-	y = Math.min(Math.ceil(rect.top + rect.height / 2), window.innerHeight - 1);
 
-	return { x, y };
-};
 /**
- * Get relevant stacks of block and inline elements, excluding line breaks
- * @method getRectStack
- * @memberof axe.commons.color
- * @instance
- * @param {Element} elm
- * @return {Array}
+ * Calculate alpha transparency of a background element obscuring the current node
+ * @private
+ * @param {Number} elmIndex
+ * @param {Array} elmStack
+ * @param {Element} originalElm
+ * @return {Number|undefined}
  */
-color.getRectStack = function(elm) {
-	let boundingCoords = color.getCoords(elm.getBoundingClientRect());
-	if (!boundingCoords) {
-		return null;
-	}
-
-	let boundingStack = dom.shadowElementsFromPoint(
-		boundingCoords.x,
-		boundingCoords.y
-	);
-
-	let rects = Array.from(elm.getClientRects());
-	// If the element does not have multiple rects, like for display:block, return a single stack
-	if (!rects || rects.length <= 1) {
-		return [boundingStack];
-	}
-
-	// Handle inline elements spanning multiple lines to be evaluated
-	let filteredArr = rects
-		.filter(rect => {
-			// exclude manual line breaks in Chrome/Safari
-			return rect.width && rect.width > 0;
-		})
-		.map(rect => {
-			let coords = color.getCoords(rect);
-			if (coords) {
-				return dom.shadowElementsFromPoint(coords.x, coords.y);
-			}
-		});
-
-	if (filteredArr.some(stack => stack === undefined)) {
-		// Can be happen when one or more of the rects sits outside the viewport
-		return null;
-	}
+function calculateObscuringAlpha(elmIndex, elmStack, originalElm) {
+	var totalAlpha = 0;
 
-	// add bounding client rect stack for comparison later
-	filteredArr.splice(0, 0, boundingStack);
-	return filteredArr;
-};
-/**
- * Get filtered stack of block and inline elements, excluding line breaks
- * @method filteredRectStack
- * @memberof axe.commons.color
- * @instance
- * @param {Element} elm
- * @return {Array}
- */
-color.filteredRectStack = function(elm) {
```

Comment:
This is a breaking change, please put this back on the `color` namespace. Apologies, I missed this in my first review round.